sirepo/smtp.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+u"""SMTP connection to send emails
+
+:copyright: Copyright (c) 2018-2019 RadiaSoft LLC.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+
+from __future__ import absolute_import, division, print_function
+from pykern.pkcollections import PKDict
+import email
+import smtplib
+
+
+class SMTP(PKDict):
+    def __init__(*args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.smtp_port = 587
+
+    def send(self, subject, sender, recipient, body):
+        if isinstance(sender, tuple):
+            sender = f'{sender[0]} <{sender[1]}>'
+        s = None
+        try:
+            # TODO(e-carlin): Understand line endings issue. EmailMessage
```

Comment:
I think [send always inserts `\r\n`](https://github.com/python/cpython/blob/3.8/Lib/smtplib.py#L968). `flatten` (in the referenced code) has to parse everything anyway.

Reply:
Nice, thx. https://github.com/radiasoft/sirepo/pull/2677/commits/5562b8e45c4790e870646d055a019adb1ceb5dc1